File: uoj_352_seasonal-war.py
# Simply need to check for adjacency between the matrix elements
# (I can't see how this is a graph)



# Algorithm: First things first: Storing the image Create a bool
# matrix of the same dim, defaults to false If a neigbor exists, set
# that value to true, and if no neighbors with True already as their
# value, increase the eagle count

case = 1

while True:
    try:
        lines = int(input())
    except EOFError:
        break

    eagles = 0
    image = []
    for i in range(lines):
        image.append(input())
    
    # Create the bool 
    for line in range(lines):
        for character in range(lines):
            if image[line][character] == '1':
                eagles += 1
                image[line][character] = '2'
            elif (image[max(0,line-1)][character] == '2'or \
                  image[min(lines-1, line+1)][character]== '2' or \
                  image[line][max(0,character-1)]== '2' or \
                  image[line][min(lines-1,character+1)] == '2'):
                    if image[line][character] = '0':
                        image[line][character] = '3'
                    if image[line][character] = '0':
                        image[line][character] = '3'
            elif image[line][character] == '1':
                eagles +=1

    # Print Image
    print("image")
    for line in range(lines):
        print(image[line])
    # Print Matrix
    print("matrix")
    for line in range(lines):
        print(boolMatrix[line])
        
    print(f"Image number {case} contains {eagles} war eagles.")
    case += 1

File: uoj_352_seasonal-war.cpp
#include <cstdio>
#include <queue>
#include <utility>

using namespace std;

int k, eaglesCounter;
char matrix[25][25];
pair<int, int> cords;


int main() {
  int cases = 1;
  while (scanf("%d", &k) != EOF) {
    for (int line = 0; line < k; line++)
      scanf("%s", matrix[line]);

    eaglesCounter = 0;
    queue<pair<int, int> > myQ;

    for (int line = 0; line < k; line++) {
      for (int index = 0; index < k; index++)
	if (matrix[line][index] == '1') {
	  eaglesCounter++;
	  myQ.push(make_pair(line, index));

	  while (not myQ.empty()) {
	    cords = myQ.front();
	    matrix[cords.first][cords.second] = '0';
	    myQ.pop();

	    for (int i = -1; i < 2; i++)
	      for (int l = -1; l < 2; l++) {
		if (((cords.first+i > k) or (cords.first+i<0)) or ((cords.second+l > k) or (cords.second+l<0)))
		  continue;
		if (matrix[cords.first+i][cords.second+l] == '1') 
		  myQ.push(make_pair(cords.first+i,cords.second+ l));
	      }
	  }
        }
    }
      printf("Image number %d contains %d war eagles.\n", cases++,
	     eaglesCounter);
  }
}
